Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -286,7 +286,11 @@ final class DeclarationTests: XCTestCase {
       fileprivate fileprivate(set) var fileprivateProp = 0
       private private(set) var privateProp = 0
       internal(set) var defaultProp = 0
-      """
+      """,
+      diagnostics: [
```

Comment:
`open open(set) var openProp = 0` is valid isn't it? Pointless, but valid 😅?

Reply:
Yes, I agree that it should be valid but it’s rejected with the same error in the C++ parser.

Also, related, I would imagine that

```swift
func foo() {
  var open = 1
  open 
  var myVar = 2
}
```

is valid but it’s rejected in the C++ parser and the safest thing I can think of doing is to also reject it in the new parser. Any new heuristics to disambiguate `open` as an identifier vs decl modifier might introduce new source breakage.

Reply:
Ah. It fails top level but doesn't fail when it's a member. Ie.
```
class Foo {
  open open(set) var openProp = 0
}
```
is fine.